File: src/MistwoodTales.Game/MoveableEntity.cpp
#include "MoveableEntity.h"
#include "Map.h"
#include "World.h"



MoveableEntity::MoveableEntity()
{
	
}


MoveableEntity::~MoveableEntity()
{
}


int MoveableEntity::Move(Direction direction)
{
	Map* map = World::Instance().CurrentMap;
	int newX, newY;
	switch (direction)
	{
	case Left:
		newX = X - 1;
		newY = Y;
		break;
	case Right:
		newX = X + 1;
		newY = Y;
		break;
	case Up:
		newX = X;
		newY = Y - 1;
		break;
	case Down:
		newX = X;
		newY = Y + 1;
		break;
	default:
		break;
	}
	if (!map->CheckIfCoordsAreAvailable(newX, newY))
		return -1;
	X = newX;
	Y = newY;
	FaceDirection = direction;
}

File: src/MistwoodTales.Game.Client/Legacy/MoveableEntity.cs
namespace MistwoodTales.Game.Client.Legacy
{
    public class MoveableEntity: Entity
    {
        public Direction FaceDirection { get; set; }

        public virtual void Move(Direction direction)
        {
            var map = World.Instance.Map;
            int newX = 0, newY = 0;
            switch (direction)
            {
                case Direction.Left:
                    newX = Point.X - 1;
                    newY = Point.Y;
                    break;
                case Direction.Right:
                    newX = Point.X + 1;
                    newY = Point.Y;
                    break;
                case Direction.Up:
                    newX = Point.X;
                    newY = Point.Y - 1;
                    break;
                case Direction.Down:
                    newX = Point.X;
                    newY = Point.Y + 1;
                    break;
                default:
                    break;
            }
            if (!map.CheckIfCoordsAreAvailable(newX, newY))
                return;
            Point.X = newX;
            Point.Y = newY;
            FaceDirection = direction;
        }

    }
}
